qtglean/include/glean/uuid.h

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef UUID_H
+#define UUID_H
+#include <QObject>
+
+#include "errortype.h"
+
+class UuidMetric final {
+  Q_GADGET
+
+ public:
+  // QML custom types require these three declarations.
+  // See: https://doc.qt.io/qt-6/custom-types.html#creating-a-custom-type
+  UuidMetric() = default;
+  UuidMetric(const UuidMetric&) = default;
+  UuidMetric& operator=(const UuidMetric&) = default;
+
+  explicit UuidMetric(int aId);
+  ~UuidMetric() = default;
+
+  Q_INVOKABLE void set(const QString& uuid) const;
+  Q_INVOKABLE QString generateAndSet() const;
+
+  // Test  only functions
+  Q_INVOKABLE QString testGetValue(const QString& pingName = "") const;
+  Q_INVOKABLE int32_t testGetNumRecordedErrors(ErrorType errorType) const;
+
+ private:
+  int m_id;
```

Comment:
Set a default value: `= 0;`

Reply:
I would argue we shouldn't do that. 1. Instances of UuidMetric objects should only be created by the glean_parser generated code through the constructor that require an id, so it will always havea value. 2. We are not doing this is any of the other metric implementations.